pkg/component/kubernetes/apiserver/authentication.go

```diff
@@ -44,90 +45,115 @@ func (k *kubeAPIServer) reconcileConfigMapAuthenticationConfig(ctx context.Conte
-		authenticationConfiguration.JWT[0].ClaimMappings.Username.Prefix = ptr.To("")
-	case len(usernamePrefix) == 0 && authenticationConfiguration.JWT[0].ClaimMappings.Username.Claim != "email":
-		authenticationConfiguration.JWT[0].ClaimMappings.Username.Prefix = ptr.To(fmt.Sprintf("%s#", authenticationConfiguration.JWT[0].Issuer.URL))
+		jwts[0].ClaimMappings.Username.Prefix = ptr.To("")
+	case len(usernamePrefix) == 0 && jwts[0].ClaimMappings.Username.Claim != "email":
+		jwts[0].ClaimMappings.Username.Prefix = ptr.To(fmt.Sprintf("%s#", jwts[0].Issuer.URL))
 	default:
-		authenticationConfiguration.JWT[0].ClaimMappings.Username.Prefix = ptr.To(usernamePrefix)
+		jwts[0].ClaimMappings.Username.Prefix = ptr.To(usernamePrefix)
 	}
 
 	if v := oidc.GroupsClaim; v != nil {
-		authenticationConfiguration.JWT[0].ClaimMappings.Groups.Claim = *v
-		authenticationConfiguration.JWT[0].ClaimMappings.Groups.Prefix = ptr.To(ptr.Deref(oidc.GroupsPrefix, ""))
+		jwts[0].ClaimMappings.Groups.Claim = *v
+		jwts[0].ClaimMappings.Groups.Prefix = ptr.To(ptr.Deref(oidc.GroupsPrefix, ""))
 	}
 
 	for key, value := range oidc.RequiredClaims {
 		claimValidationRule := apiserverv1beta1.ClaimValidationRule{
 			Claim:         key,
 			RequiredValue: value,
 		}
-		authenticationConfiguration.JWT[0].ClaimValidationRules = append(authenticationConfiguration.JWT[0].ClaimValidationRules, claimValidationRule)
-	}
-
-	data, err := runtime.Encode(ConfigCodec, authenticationConfiguration)
-	if err != nil {
-		return "", fmt.Errorf("unable to encode authentication configuration: %w", err)
+		jwts[0].ClaimValidationRules = append(jwts[0].ClaimValidationRules, claimValidationRule)
 	}
 
-	return string(data), nil
+	return jwts
 }
 
 func (k *kubeAPIServer) handleAuthenticationSettings(deployment *appsv1.Deployment, configMapAuthenticationConfig *corev1.ConfigMap, secretOIDCCABundle *corev1.Secret) {
-	if value, ok := k.values.FeatureGates["StructuredAuthenticationConfiguration"]; versionutils.ConstraintK8sLess130.Check(k.values.Version) || (ok && !value) {
+	var (
+		structAuthGateEnabled, structAuthGateSet  = k.values.FeatureGates["StructuredAuthenticationConfiguration"]
+		structAuthIsDisabled                      = (structAuthGateSet && !structAuthGateEnabled)
+		anonymousGateEnabled, anonymousGateSet    = k.values.FeatureGates["AnonymousAuthConfigurableEndpoints"]
+		shouldConfigureAnonymousViaStructuredAuth = (!anonymousGateSet || anonymousGateEnabled) && versionutils.ConstraintK8sGreaterEqual132.Check(k.values.Version)
+	)
+
+	if versionutils.ConstraintK8sLess130.Check(k.values.Version) || structAuthIsDisabled {
 		k.handleOIDCSettings(deployment, secretOIDCCABundle)
-		return
+	}
+
+	if !shouldConfigureAnonymousViaStructuredAuth || structAuthIsDisabled {
+		deployment.Spec.Template.Spec.Containers[0].Args = append(deployment.Spec.Template.Spec.Containers[0].Args, "--anonymous-auth="+strconv.FormatBool(ptr.Deref(k.values.AnonymousAuthenticationEnabled, false)))
```

Comment:
IIUC, this flag would be set for `1.31` clusters that configure anonymous-auth via the `AuthenticationConfiguration`. Can you double check? WDYT adding a test for this scenario as well?

Reply:
You are right! I now do a more practical check https://github.com/gardener/gardener/pull/12198/commits/574b089e85e82ae75034de463f88a5978f92dc85 which I think works better. PTAL